app/models/operatingsystem.rb

```diff
@@ -255,8 +255,18 @@ def boot_filename(host = nil)
     return default_boot_filename if host.nil? || host.pxe_loader.nil?
     return host.foreman_url('iPXE') if host.pxe_loader == 'iPXE Embedded'
     architecture = host.arch.nil? ? '' : host.arch.bootfilename_efi
-    boot_uri = URI.parse((host.subnet&.httpboot?) ? host.subnet.httpboot.url : Setting[:unattended_url])
-    self.class.all_loaders_map(architecture, "#{boot_uri.host}:#{boot_uri.port}")[host.pxe_loader]
+    if host.subnet&.httpboot?
+      if host.pxe_loader =~ /UEFI HTTPS/
+        port = host.subnet.httpboot.setting(:HTTPBoot, 'https_port') || raise(::Foreman::Exception.new(N_("HTTPS boot requires proxy with httpboot feature and https_port exposed setting")))
```

Comment:
One question I have is about `:HTTPBoot` vs a string. Does it matter?

Reply:
I'm wondering if it would be better to invalidate the host object (via a rails validation) in addition to throwing an exception here. The exception is not very user-friendly imho.

Reply:
> One question I have is about `:HTTPBoot` vs a string. Does it matter?

I am not following.

> invalidate the host object

Good idea, lemme see if it's easy change.

Reply:
Hmmm the code is used only in orchestration and there we tend to use exceptions as the orchestration stack tries to catch them and roll back. So I'd probably keep it as-is.

Reply:
> One question I have is about :HTTPBoot vs a string. Does it matter?

If you mean that I should use 'HTTPBoot' instead, I can change it if you insist but ActiveRecord apparently works fine with that.

Reply:
What's the outcome of this discussion? is this good to merge or do you plan to change it?

Reply:
The ball is on @ekohl side, thanks!

Reply:
What is the status @ekohl ? If there is a change you want me to do let me know and I will happily do it. This is a customer-wanted feature, thanks!

Reply:
Again, I can change the `:HTTPBoot` to string if that's what @ekohl wants, I am not sure. Please confirm.